### PR TITLE
FUNPY-26 Improves snapshots sync time

### PR DESCRIPTION
**Pull request by nessv**
# Description 
Improves the sync time for snapshots

# Changes 

- New GET call to retrieve all the snapshots asociated with a family.
- Replaces the GET request that retrieves all the families, this in order to fix a
bug that was causing to show a lot more families compare with the web application.

# Tests 

This code has been tested in the following ways:
- Tasks:
  - [X] Logged in to the application
  - [X] Fill out a snapshot for a new family
  - [X] Fill out a snapshot for an existing family
  - [ ] Fill out priorities for a snapshot
  - [X] View a family and it's priorities
- Screen Size:
  - [ ] 7" screen size
  - [X] 8" screen size
  - [ ] 10" screen size

### CODE REVIEW

File: app/src/main/java/org/fundacionparaguaya/adviserplatform/data/remote/FamilyService.java [hunk 1]
```diff
@@ -17,8 +17,8 @@
 
 public interface FamilyService {
 
-    @GET("families")
-    Call<List<FamilyIr>> getFamilies();
+    @GET("families/user")
+    Call<List<FamilyIr>> getFamilies(@Query("name") String name);
```
Comment (alefq) [inline app/src/main/java/org/fundacionparaguaya/adviserplatform/data/remote/FamilyService.java]: This should keep uniformity with API Path and FP-PSP-SERVER names. 
Rename the method to `getFamiliesByUserAndNameOrCode`, and `name` parameter to `nameOrCode`.

File: app/src/main/java/org/fundacionparaguaya/adviserplatform/data/repositories/SnapshotRepository.java [hunk 2]
```diff
@@ -278,32 +278,23 @@ private boolean pullSnapshots() {
...
-        for (Family family : families) {
-            for (Survey survey : surveysNow) {
 
-                if(shouldAbortSync()) return false;
-
-                //TODO Sodep: time complexity n^2
-                Log.d(TAG, stopWatch.lap(String.format("Family: %s, Survey: %s", family.getName(),
-                        survey.getDescription())));
-                success &= pullSnapshots(family, survey);
-                if(success && getDashActivity() != null) {
```
Comment (alefq) [inline app/src/main/java/org/fundacionparaguaya/adviserplatform/data/repositories/SnapshotRepository.java:290]: Restore this `DashActivity` progress indicator maintaining coherence between `recordsCount` from `BaseRepository` and local `loopCount`

File: app/src/main/java/org/fundacionparaguaya/adviserplatform/data/repositories/SnapshotRepository.java [hunk 3]
```diff
@@ -346,7 +337,7 @@ private boolean pullSnapshots(Family family, Survey survey)
                         throw new HttpException(prioritiesResponse);
                     }
                     snapshot.setPriorities(
-                            IrMapper.mapPriorities(prioritiesResponse.body(), survey));
+                            IrMapper.mapPriorities(prioritiesResponse.body(), surveyList.get(snapshot.getSurveyId() - 1)));
```
Comment (alefq) [inline app/src/main/java/org/fundacionparaguaya/adviserplatform/data/repositories/SnapshotRepository.java]: This causes errors on Test Server

```
06-22 14:30:51.045 10626-10960/org.fundacionparaguaya.adviserplatform E/SyncManager: sync: Error while syncing!
    java.lang.IndexOutOfBoundsException: Index: 19, Size: 5
        at java.util.ArrayList.get(ArrayList.java:437)
        at org.fundacionparaguaya.adviserplatform.data.repositories.SnapshotRepository.pullSnapshots(SnapshotRepository.java:341)
        at org.fundacionparaguaya.adviserplatform.data.repositories.SnapshotRepository.pullSnapshots(SnapshotRepository.java:285)
        at org.fundacionparaguaya.adviserplatform.data.repositories.SnapshotRepository.sync(SnapshotRepository.java:371)
        at org.fundacionparaguaya.adviserplatform.data.repositories.BaseRepository.sync(BaseRepository.java:39)
        at org.fundacionparaguaya.adviserplatform.data.repositories.SyncManager.resyncWithOneAuthentication(SyncManager.java:189)
        at org.fundacionparaguaya.adviserplatform.data.repositories.SyncManager.sync(SyncManager.java:136)
        at org.fundacionparaguaya.adviserplatform.jobs.SyncJob.onRunJob(SyncJob.java:62)
        at com.evernote.android.job.Job.runJob(Job.java:132)
        at com.evernote.android.job.JobExecutor$JobCallable.runJob(JobExecutor.java:191)
        at com.evernote.android.job.JobExecutor$JobCallable.call(JobExecutor.java:176)
        at com.evernote.android.job.JobExecutor$JobCallable.call(JobExecutor.java:159)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
        at java.lang.Thread.run(Thread.java:764)
```

**Review by alefq — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/fundacionparaguaya/adviserplatform/data/remote/FamilyService.java (hunk 1 above)
Comment (nessv) [reply app/src/main/java/org/fundacionparaguaya/adviserplatform/data/remote/FamilyService.java]: I will change the method name as well as the variable name. But on the server side is expecting `name`, so I will leave `@Query("name")` as it is.

File: app/src/main/java/org/fundacionparaguaya/adviserplatform/data/repositories/SnapshotRepository.java (hunk 2 above)
Comment (nessv) [reply app/src/main/java/org/fundacionparaguaya/adviserplatform/data/repositories/SnapshotRepository.java:290]: I will restore `DashActivity` as requested 👌🏻

File: app/src/main/java/org/fundacionparaguaya/adviserplatform/data/repositories/SnapshotRepository.java (hunk 3 above)
Comment (nessv) [reply app/src/main/java/org/fundacionparaguaya/adviserplatform/data/repositories/SnapshotRepository.java]: I'm not been able to reproduce it. Under what circumstances did you get it?

**Comment by nessv** [comment]
@alefq I made a new commit with the changes you requiere. I wasn't able to reproduce the error you pointed up (I've tested locally as well as in Test enviorment). Please can you confirm that the error persists?

**Pull request merged**